File: be/src/main/java/com/quangnt0000/be_modul/modal/DataLake/Employee.java
package com.quangnt0000.be_modul.modal.DataLake;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@EntityListeners(AuditingEntityListener.class)
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String name;
    private String email;

    @Column(unique = true, nullable = false)
    private String phone;

    private String address;
    private LocalDate birthday;
    private String gender;

    private String position;

    private String keyAvatar;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;

    @Builder.Default
    private Boolean deleted = false;

    @ManyToMany
    @JoinTable(
        name = "employee_department",
        joinColumns = @JoinColumn(name = "employee_id"),
        inverseJoinColumns = @JoinColumn(name = "department_id")
    )
    @Builder.Default
    private List<Department> departments = new ArrayList<>();
}

File: fe-modul-report/src/features/employee/types/employee.ts
export interface DepartmentSimple {
    id: string;
    name: string;
}

export interface EmployeeResponse {
    id: string;
    name: string;
    departments: DepartmentSimple[];
    position: string;
    phone: string;
    email: string;
    address: string;
    gender: string;
    birthday: string;
    keyAvatar: string | null;
    role: string | null;
}

export interface EmployeeRequest {
    id?: string | null;
    name: string;
    email: string;
    phone: string;
    address: string;
    birthday: string;
    gender: string;
    position: string;
    departmentIds: string[]; 
    avatarFile?: File | null;
}

